Nautilus/Assets/Gadgets/GadgetExtensions.cs

```diff
@@ -107,13 +107,52 @@ public static ScanningGadget SetPdaGroupCategory(this ICustomPrefab customPrefab
         if (!customPrefab.TryGetGadget(out ScanningGadget scanningGadget))
         {
             scanningGadget = new ScanningGadget(customPrefab, TechType.None);
+            customPrefab.AddGadget(scanningGadget);
         }
 
-        scanningGadget.WithPdaGroupCategory(group, category);
+        return scanningGadget.WithPdaGroupCategory(group, category);
+    }
```

Comment:
Since we haven't really decided on PR #404, I would suggest changing this to the previous implementation of a gadget extension.